plugin/src/main/java/org/apache/jackrabbit/filevault/maven/packaging/VaultMojo.java

```diff
@@ -534,6 +489,94 @@ public void execute() throws MojoExecutionException, MojoFailureException {
+                throw new MojoExecutionException("conflicting filters.");
+            }
+            // load filters for further processing
+            filters.load(filterFile);
+
+            getLog().warn("The project is using a filter.xml provided via the resource plugin.");
+            getLog().warn("This is deprecated and might no longer be supported in future versions.");
+            getLog().warn("Use the 'filterSource' property to specify the filter or use inline filters.");
+            return;
+        }
+
+        // if last modified of vault-work/META-INF/vault/filter.xml == 0 -> delete it
+        if (filterFile.exists() && filterFile.lastModified() == 0) {
+            if (!filterFile.delete()) {
+                getLog().error("Unable to delete previously generated filter.xml. re-run the goals with a clean setup.");
+                throw new MojoExecutionException("Unable to delete file.");
+            }
+        }
+
+        // if filterSource exists, read the filters into sourceFilters
+        DefaultWorkspaceFilter sourceFilters = new DefaultWorkspaceFilter();
+        if (filterSource != null && filterSource.exists()) {
+            getLog().info("Loading filter from " + filterSource.getPath());
+            sourceFilters.load(filterSource);
+            if (!filters.getFilterSets().isEmpty()) {
+                getLog().info("Merging inline filters.");
+                sourceFilters.merge(filters);
+            }
+            filters.getFilterSets().clear();
+            filters.getFilterSets().addAll(sourceFilters.getFilterSets());
+
+            // reset source filters for later. this looks a bit complicated but is needed to keep the same
+            // filter order as in previous versions
+            sourceFilters = new DefaultWorkspaceFilter();
+            sourceFilters.load(filterSource);
+            sourceFilters.generateSource();
+        }
+
+        // if the prefix property is set, it should be used if no filter is set
+        if (filters.getFilterSets().isEmpty() && prefix.length() > 0) {
+            addWorkspaceFilter(prefix);
+        }
+
+        // if no filter is defined at all, fail
+        if (filters.getFilterSets().isEmpty()) {
+            if (failOnEmptyFilter) {
+                final String msg = "No workspace filter defined (failOnEmptyFilter=true)";
+                getLog().error(msg);
+                throw new MojoExecutionException(msg);
+            } else {
+                getLog().warn("No workspace filter defined. Package import might have unexpected results.");
+            }
+        }
+
+        // if the source filters and the generated filters are the same, copy the source file to retain the comments
+        if (filterSource != null && sourceFilters.getSourceAsString().equals(filters.getSourceAsString())) {
+            FileUtils.copyFile(filterSource, filterFile);
+        } else {
+            // generate xml and write to filter.xml
```

Comment:
Add a log line line here similar to line 534 "Generating filter.xml from plugin configuration"